recipes/glfw/all/conanfile.py

```diff
@@ -23,49 +27,85 @@ class GlfwConan(ConanFile):
         "shared": [True, False],
         "fPIC": [True, False],
         "vulkan_static": [True, False],
+        "with_x11": [True, False],
+        "with_wayland": [True, False],
     }
     default_options = {
         "shared": False,
         "fPIC": True,
         "vulkan_static": False,
+        "with_x11": True,
+        "with_wayland": False,
     }
 
+    @property
+    def _has_build_profile(self):
+        return hasattr(self, "settings_build")
+
     def export_sources(self):
         export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
+        if self.settings.os != "Linux":
+            self.options.rm_safe("with_wayland")
+        if not self.settings.os in ["Linux", "FreeBSD"] or self.version <= Version("3.3.8"):
+            self.options.rm_safe("with_x11")
 
     def configure(self):
         if self.options.shared:
             self.options.rm_safe("fPIC")
         self.settings.rm_safe("compiler.cppstd")
         self.settings.rm_safe("compiler.libcxx")
 
+        if self.options.get_safe("with_wayland"):
+            self.options["xkbcommon"].with_wayland = True
+
     def layout(self):
         cmake_layout(self, src_folder="src")
 
     def requirements(self):
         self.requires("opengl/system")
         if self.options.vulkan_static:
-            self.requires("vulkan-loader/1.3.239.0")
-        if self.settings.os in ["Linux", "FreeBSD"]:
+            self.requires("vulkan-loader/1.3.243.0")
+        if self.options.get_safe("with_x11") or self.version <= Version("3.3.8"):
```

Comment:
This made it impossible to build glfw on windows, because it always requires xorg